js/interface.js

```diff
@@ -143,7 +143,7 @@ function initDataSourceProvider(currentDataSourceId) {
           $dataColumnsPass.val(data.passColumn);
         }
 
-        currentDataSource = dataSource;
+        currentDataSource = dataSource.id ? dataSource : null;
```

Comment:
does `dataSource` always exist? otherwise, please add a `dataSource &&` prefix

Reply:
@squallstar Yes, it always exists because this is an incoming parameter in the `onEvent` function within the data source provider.